langchain/vectorstores/weaviate.py

```diff
@@ -124,32 +124,31 @@ def add_texts(
         metadatas: Optional[List[dict]] = None,
         **kwargs: Any,
     ) -> List[str]:
-        """Upload texts with metadata (properties) to Weaviate."""
-        from weaviate.util import get_valid_uuid
-
+        """
+        Upload texts with metadata (properties) to Weaviate.
+        Pass "uuids" list to set custom IDs or overwrite existing texts
+        Pass "vectors" list to use custom vectors for the texts
+        """
         ids = []
+        if not isinstance(texts, list):
+            texts = list(texts)
+
+        vectors: Optional[List[List[float]]] = kwargs.get("vectors")
```

Comment:
lets not put this logic here - if you want to upload embeddings directly, we should create an `add_embeddings` method like we do in some other vectorstores

Reply:
Yeah that makes sense.

It would be nice if add_embeddings was actually a part of the VectorStore interface. I see slightly different versions of it in faiss and pgvector. Ideally something like
```
    def add_embeddings(
        self,
        texts: List[str],
        embeddings: List[List[float]],
        metadatas: Optional[List[dict]] = None,
        **kwargs: Any,
    ) -> List[str]:
```

Wdyt? Guess it's partially a question of whether or not all the vector stores would support this, but I can't imagine they wouldn't.

Reply:
Actually it's less code to maintain overall if you add `embeddings: Optional[List[List[float]]] = None` to the `add_texts` interface. Because with the add_texts/add_embeddings style you end up either duplicating code (e.g. `PGVector`) or creating a third private method to handle the real add operation (e.g. `FAISS.__add`)

Reply:
i think we should standardize that interface, but probably out of scope of this pr

Reply:
https://github.com/hwchase17/langchain/issues/5945